torch/csrc/api/src/nn/modules/pooling.cpp

```diff
@@ -273,6 +273,91 @@ template class MaxUnpoolImpl<3, MaxUnpool3dImpl>;
 
 // ============================================================================
 
+FractionalMaxPool2dImpl::FractionalMaxPool2dImpl(const FractionalMaxPool2dOptions& options_)
+    : options(options_) {
+  reset();
+}
+
+void FractionalMaxPool2dImpl::reset() {
+  _random_samples = register_buffer("_random_samples", options._random_samples());
+  if (options.output_size() == c10::nullopt && options.output_ratio() == c10::nullopt) {
```

Comment:
I don't insist, but WDYT about this?
```suggestion
  if (!options.output_size() && !options.output_ratio()) {
```

Reply:
I think either way works - I was using `== c10::nullopt` because it looks more similar to Python version and also enforces the type of `options.output_size()` more strongly, in the sense that if we ever change the type of `options.output_size()` to not be `c10::optional<...>`, `options.output_size() == c10::nullopt` would fail to compile, while `!options.output_size()` might still compile (because the new type might also overload the `bool()` operator). In such a scenario, failing to compile is preferred over being able to compile, because we won't be sure if the new type's `bool()` operator is checking the right thing that we care about at the original call site.